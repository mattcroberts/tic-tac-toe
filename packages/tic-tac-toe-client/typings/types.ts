/* tslint:disable */
import { GraphQLResolveInfo } from "graphql";

export type Resolver<Result, Parent = any, Context = any, Args = any> = (
    parent: Parent,
    args: Args,
    context: Context,
    info: GraphQLResolveInfo
) => Promise<Result> | Result;

export type SubscriptionResolver<
    Result,
    Parent = any,
    Context = any,
    Args = any
> = {
    subscribe<R = Result, P = Parent>(
        parent: P,
        args: Args,
        context: Context,
        info: GraphQLResolveInfo
    ): AsyncIterator<R | Result>;
    resolve?<R = Result, P = Parent>(
        parent: P,
        args: Args,
        context: Context,
        info: GraphQLResolveInfo
    ): R | Result | Promise<R | Result>;
};

export interface Query {
    grid: Grid;
}

export interface Grid {
    id: string;
    gridItems: GridItem[][];
    currentPlayer: Player;
    winner?: Player | null;
    isFinished: boolean;
    size: number;
}

export interface GridItem {
    id: string;
    player?: Player | null;
}

export interface Mutation {
    executeTurn: Grid;
    newGame: Grid;
}
export interface GridQueryArgs {
    id: string;
}
export interface ExecuteTurnMutationArgs {
    id: string;
    player: string;
    x: number;
    y: number;
}

export enum Player {
    NAUGHT = "NAUGHT",
    CROSS = "CROSS"
}

export namespace QueryResolvers {
    export interface Resolvers<Context = any> {
        grid?: GridResolver<Grid, any, Context>;
    }

    export type GridResolver<R = Grid, Parent = any, Context = any> = Resolver<
        R,
        Parent,
        Context,
        GridArgs
    >;
    export interface GridArgs {
        id: string;
    }
}

export namespace GridResolvers {
    export interface Resolvers<Context = any> {
        id?: IdResolver<string, any, Context>;
        gridItems?: GridItemsResolver<GridItem[][], any, Context>;
        currentPlayer?: CurrentPlayerResolver<Player, any, Context>;
        winner?: WinnerResolver<Player | null, any, Context>;
        isFinished?: IsFinishedResolver<boolean, any, Context>;
        size?: SizeResolver<number, any, Context>;
    }

    export type IdResolver<R = string, Parent = any, Context = any> = Resolver<
        R,
        Parent,
        Context
    >;
    export type GridItemsResolver<
        R = GridItem[][],
        Parent = any,
        Context = any
    > = Resolver<R, Parent, Context>;
    export type CurrentPlayerResolver<
        R = Player,
        Parent = any,
        Context = any
    > = Resolver<R, Parent, Context>;
    export type WinnerResolver<
        R = Player | null,
        Parent = any,
        Context = any
    > = Resolver<R, Parent, Context>;
    export type IsFinishedResolver<
        R = boolean,
        Parent = any,
        Context = any
    > = Resolver<R, Parent, Context>;
    export type SizeResolver<
        R = number,
        Parent = any,
        Context = any
    > = Resolver<R, Parent, Context>;
}

export namespace GridItemResolvers {
    export interface Resolvers<Context = any> {
        id?: IdResolver<string, any, Context>;
        player?: PlayerResolver<Player | null, any, Context>;
    }

    export type IdResolver<R = string, Parent = any, Context = any> = Resolver<
        R,
        Parent,
        Context
    >;
    export type PlayerResolver<
        R = Player | null,
        Parent = any,
        Context = any
    > = Resolver<R, Parent, Context>;
}

export namespace MutationResolvers {
    export interface Resolvers<Context = any> {
        executeTurn?: ExecuteTurnResolver<Grid, any, Context>;
        newGame?: NewGameResolver<Grid, any, Context>;
    }

    export type ExecuteTurnResolver<
        R = Grid,
        Parent = any,
        Context = any
    > = Resolver<R, Parent, Context, ExecuteTurnArgs>;
    export interface ExecuteTurnArgs {
        id: string;
        player: string;
        x: number;
        y: number;
    }

    export type NewGameResolver<
        R = Grid,
        Parent = any,
        Context = any
    > = Resolver<R, Parent, Context>;
}
