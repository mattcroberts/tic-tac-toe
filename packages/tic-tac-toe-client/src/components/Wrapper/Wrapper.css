.root {
    padding: 6px;
    height: 100vh;
}

.top {
    display: flex;
    align-items: center;
}

.top > * {
    margin-right: 6px;
}

.heading {
    margin: 6px 6px 6px 0;
}

.button {
    composes: button from "../../pages/Home/Home.css";
}

.turnInfo > span {
    margin-right: 6px;
}
