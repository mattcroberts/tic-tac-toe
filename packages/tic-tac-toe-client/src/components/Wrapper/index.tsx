import * as React from "react";

import { Link } from "react-router-dom";
import style from "./Wrapper.css";
import Button, { Size } from "../Button";
import GridControls from "../GridControls";
import { Player as IPlayer } from "../../../typings/types";
import getPlayerSymbol from "../../utils/playerMap";

export default ({
    children,
    controllingPlayer,
    currentPlayer,
    isMultiplayer,
    gameUrl
}: {
    children: React.ReactNode;
    controllingPlayer: IPlayer;
    currentPlayer: IPlayer;
    isMultiplayer: boolean;
    gameUrl: string;
}) => (
    <div className={style.root}>
        <div>
            <div className={style.top}>
                <h1 className={style.heading}>Tic Tac Toe</h1>
                <Link to="/">
                    <Button className={style.button} size={Size.SMALL}>
                        Home
                    </Button>
                </Link>
                <GridControls isMultiplayer={isMultiplayer} gameUrl={gameUrl} />
            </div>
            <div className={style.turnInfo}>
                <span>
                    You:{" "}
                    {getPlayerSymbol(controllingPlayer.symbol).toUpperCase()}
                </span>
                <span>
                    Turn: {getPlayerSymbol(currentPlayer.symbol).toUpperCase()}
                </span>
            </div>
        </div>
        <div className={style.content}>{children}</div>
    </div>
);
