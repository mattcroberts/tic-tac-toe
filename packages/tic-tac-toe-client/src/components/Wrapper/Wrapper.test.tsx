import * as React from "react";
import { shallow } from "enzyme";
import Wrapper from ".";
import { Symbol as ISymbol } from "../../../typings/types";

describe("Wrapper", () => {
    const defaultProps = {
        children: <div>Test</div>,
        controllingPlayer: { id: "123", symbol: ISymbol.CROSS },
        currentPlayer: { id: "123", symbol: ISymbol.CROSS }
    };

    it("should render", () => {
        const component = shallow(<Wrapper {...defaultProps} />);
        expect(component).toMatchSnapshot();
    });
});
