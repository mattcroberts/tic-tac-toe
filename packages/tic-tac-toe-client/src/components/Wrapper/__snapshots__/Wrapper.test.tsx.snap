// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Wrapper should render 1`] = `
<div
  className="root"
>
  <div
    className="inner"
  >
    <div
      className="top"
    >
      <h1
        className="heading"
      >
        TIC TAC TOE
      </h1>
      <Link
        className="link"
        replace={false}
        to="/"
      >
        Home
      </Link>
    </div>
    <div
      className="content"
    >
      <div>
        Test
      </div>
    </div>
  </div>
</div>
`;
